dependency/retrieval/main.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"flag"
+	"log"
+	"os"
+
+	"github.com/paketo-buildpacks/icu/dependency/retrieval/components"
+	"github.com/paketo-buildpacks/packit/v2/cargo"
+)
```

Comment:
Rob brought up a good point on another dependency PR:

> Some of the other additions of dependency logic have a separate go.mod under the dependency directory. This is good for a couple of reasons:

>The buildpack itself doesn't pull in libraries that are only used by the dependency code
It is more obvious to humans/tooling which libraries are used by the buildpack vs the dependency code.
Do you have any thoughts on this? I'd like to split them if possible.

I'm in agreement with him on that, if you're open to making this change so the top-level go.mod can stay clean

Reply:
I am having issues adding a `go.mod` files to the retrieval directory because it requires things from the components package which has not been released yet. I would have to rework the structure of the package entirely. I can take a look and see what I can do.

Reply:
Interesting. I structure my code the same way as you in MRI and didn't run into that: https://github.com/paketo-buildpacks/mri/tree/main/dependency/retrieval

Reply:
I would have to completely rearrange the structure of retrieval to accommodate for this change. Currently this will not work because `main` requires packages from `github.com/paketo-buildpacks/icu/dependency/retrieval/components` which has not been released yet because it is part of this PR.

I can tinker with this some more if you think it is valuable for the first release but this is also something that we could address later if we come up with a more uniform way of writing these actions as we refactor them going forward.

Reply:
Sorry to harp on this, I think its worth doing if we can. I pushed up what I had in mind to a branch based off this one: https://github.com/paketo-buildpacks/icu/commit/a43156e14719f5a20384bc659a40a642a02eae14

Let me know if there's an issue with this I'm not seeing

Reply:
The only problem I ran into was that anchore's `packageurl` package was getting pulled in rather than package-url

Reply:
If you just want to pr that change in on top of this branch I would be happy to test it a merge it.

Reply:
done see #335